### ISSUE TITLE
Sort property for sort-order in base element is not generated correctly

### ISSUE THREAD

**Issue by eehondas** [dnadesign/silverstripe-elemental#499]
Hi,
The sort order on the BaseElement.php is not generated correctly.
```php
    public function onBeforeWrite()
    {
        parent::onBeforeWrite();

        if (!$this->Sort) {
            $this->Sort = static::get()->max('Sort') + 1;
        }
    }
```
This piece of code itself is correct, but it only accounts for Elements of the same 'type'.
static gets resolved to the extended Elements className.
For example let's say I have 1 Element of type A, then 4 Elements of type B and I create a new Element of type A, I'm expecting the new Element on position 6, in reality it is in position 2 and shares the same Sort index as the first Element of type B because those were not considered.
The fix is as easy as this (every new element extends BaseElement anyway):
```php
    public function onBeforeWrite()
    {
        parent::onBeforeWrite();

        if (!$this->Sort) {
            $this->Sort = BaseElement::get()->max('Sort') + 1;
        }
    }
```

Thanks

**Comment by robbieaverill** [comment]
Good point. Would you like to make a PR?

**Comment by eehondas** [comment]
Yeah sure.
I'm actually forking. Should I work from branch 3?

**Comment by ScopeyNZ** [comment]
> I'm actually forking. Should I work from branch 3?

Yes please `3.0` branch would be best. We can merge up the branches once you're done 🙂

**Comment by eehondas** [comment]
#513

**Comment by robbieaverill** [comment]
Looks like this change was already made in the 3.1 release line =)
